### ISSUE TITLE
Use new gotenberg pdfa options

### ISSUE THREAD

**Issue by acn128** [stumpylog/gotenberg-client#5]
Hi,

I hope I'm in the right place with this request.

When importing a LibreOffice document into paperless-ngx, it gets converted by gotenberg to PDF.
But the resulting PDF file does sometimes have formatting problems, eg. the right alignment in a table of contents is wrong.

I've opened an issue at gotenberg: https://github.com/gotenberg/gotenberg/issues/728

It seems that paperless-ngx is calling gotenberg via old (and deprecated) options: (see: https://github.com/gotenberg/gotenberg/issues/728#issuecomment-1838414378)

> So, paperless is sending pdfFormat and not nativePdfFormat (both are deprecated btw, it should now use pdfa).
> By sending pdfFormat, Gotenberg first converts the document to PDF, then asks a PDF engine to convert it to PDF/A-2b.
> In the default version of Gotenberg, only LibreOffice can convert a document to a PDF/A format, that's why you're seeing two unoconv processes.
> Using nativePdfFormat (or better pdfa), it will convert the document to PDF + PDF/A-2b in one process.

I guess that paperless-ngx is using this package for talking to gotenberg, so I would ask if it could be possible to update the options that are used for creating PDF files from LibreOffice documents.

Thank you!

**Comment by stumpylog** [comment]
@gulien This is undfortunatly the exact situation I wanted to avoid as mentioned [here](https://github.com/gotenberg/gotenberg/issues/710#issuecomment-1818046685).  With paperless, people are highly likely to be running anything from 7.7 to 8.0 (eventually).  Without knowing what version they have, the conversion might produce wrong results, such as not actually getting PDF/A results.

Does 7.10 have a ordering to the form fields so I can send `pdfa`, `nativePdfFormat` and `pdfFormat` in an attempt to remain compatible?

**Comment by gulien** [comment]
For Gotenberg 7, using `nativePdfFormat` is the most compatible form field. Although I’d suggest to offer the option to choose to the end user and mark the deprecated form field as such. For Gotenberg 8, releasing a major version (X.y.z) with only the `pdfa` form field should do the trick.

**Comment by stumpylog** [comment]
The problem is, the end user has no idea what these options mean.  They're using the library incidentally in an application.  They might not even know what version they are using and there's no good way to force everyone to update.  I don't trust users to read the release notes carefully.

**Comment by gulien** [comment]
Fair enough. In my case I expect people to know Gotenberg first and use the client accordingly. Also, in PHP, you can mark a function as deprecated and it will show as such in your IDE. Dunno if that exits in Python.

**Comment by stumpylog** [comment]
I can mark something as deprecated, but since I develop paperless too, I kind of already know that.  And without a version check endpoint, I couldn't report something to the user suggesting they upgrade to 7.10.

Why does Gotenberg not translate from `nativePdfFormat` or `pdfFormat` to `pdfa` (with a warning) as they are basically the same option, but apparently have slightly different results (from the original issue).

**Comment by gulien** [comment]
It kind of do that already: https://github.com/gotenberg/gotenberg/blob/66142b1dd8e83265d9cb16ceed348647f9ee48db/pkg/modules/libreoffice/routes.go#L56C13-L56C13

I accept all form fields, with warnings for deprecated form fields.

They are not the sale options though:

* `pdfFormat` asks for a PDF engine to convert to a specific PDF/A format. In the default image it’s always LibreOffice, but there are custom Gotenberg variants using Ghostscript for such thing.
* `nativePdfFormat` forces the usage of LibreOffice, and it’s also more efficient because it does the conversion to PDF and PDF/A in the same process.
* `pdfa` now replaces both previous form fields, as we now also accepts the PDF/UA format alongside. There is an hidden form field `nativePdfFormats` (bool) that is set by default to true,  but may be overridden for custom variant.

**Comment by stumpylog** [comment]
Ok, yeah, so if I understand Go rightly, `pdfa` form field would override `pdfFormat` if provided, just with a warning.  Which seems like the best to give everyone time to upgrade in the meantime.

**Comment by stumpylog** [comment]
Fixed with 52426e99b9ce677ba5614c54eb952b57bce81885, both options will be sent for now.  And I'll start pushing our users towards upgrading to 7.10, before removing this compatibility fix after a reasonable amount of time

**Comment by acn128** [comment]
Thank you both very much for resolving this issue!

@stumpylog: How can I test the changes in the new gotenberg-client in paperless? I'm using this image:
`image: ghcr.io/paperless-ngx/paperless-ngx:latest`
I'm sorry, but I don't have that much experience in this docker stuff yet. Thank you!

**Comment by stumpylog** [comment]
We're going to be doing a paperless release soon, probably today, with the updated client included.  

On 7.10, you will see the warning about `pdfFormat` for now, but Gotenberg should be using the `pdfa` option which is also provided by the client

**Comment by acn128** [comment]
Thank you very much for the quick solution, it's working! *happy*
